architecture-decisions/0003-preservation.md

```diff
@@ -0,0 +1,114 @@
+      following:
+      ```
+      - <resource-id>
+        - data
+          - <child-id>
+            - <child-id>.json
+            - <binary.tif>
+        - <resource-id>.json
+      ```
+   1. Children are preserved on save if their parents are preserved.
+   1. Related objects such as collections, Ephemera Terms, etc. will not be
+      packaged inside the preserved object. If it's important they be preserved,
+      those objects should be marked with the `cloud` preservation policy.
+   1. When a FileSet is added to a resource which is already complete and marked
+      with the `cloud` preservation policy, it will upload the new binary
+      content to both the repository and to Google Cloud Storage.
+   1. If a child is marked to be preserved, but its parent is not, it will still
+      save in a nested directory structure, but will not automatically create
+      backups of its parents.
+   1. This behavior will be attached to the ChangeSetPersister.
+1. Packaging Details
+   1. When preserved a `PreservationObject` will be created in Figgy with a
+      `preserved_object_id` property which points to the object it's preserving.
+   1. Each `PreservationObject` will contain `FileMetadata` for the binary
+      object as well as a serialized JSON file of the resource it's preserving.
+      On upload to preservation, those items' checksums will be calculated and
+      stored on the `PreservationObject`.
+   1. JSON metadata will have the use `pcdm:PreservedMetadata` and binary
+      content will have the use `pcdm:PreservationCopy`
+   1. We will only keep the most recent version of any file, overwriting any
+      files which match the same file name, but relying on versioning to go back
+      if necessary.
+   1. When a preserved resource is deleted, we will delete its directory from
+      preservation storage. If we need to get it again, we will look at Google
+      Cloud Storage's stored versions.
+   1. If a child's hierarchy changes (it moves parents), we will move the
+      content in the preservation storage to match.
+   1. When a file's binary content is replaced on disk, we will upload a new
+      copy of the file to preservation and calculate a new checksum.
+2. Fixity Checks
+   1. Technical details of fixity checking will occur in a later ADR.
+   1. A random subset of the preserved copies will have their files pulled down
+      from preservation storage, their checksums calculated as they're streamed,
+      and then compared to the checksum of the object stored locally.
+   1. In the case of a failure it will be reported to Figgy and displayed in a
+      dashboard for further follow-up and repair.
+
+## Consequences
+
+1. Structure
+   1. The resultant structure will not be in a format that is expected by
+      outside vendors. However, we have a BagIt packager for those use cases,
+      and this structure can be easily converted to a BagIt bag via calculation
```

Comment:
"calculation" seems ambiguous here.  We already have the relevant checksums for the data files, right? So we could create a Bag without needing to re-checksum those (which will make it much faster and prevent missing corruption)

Reply:
K, changed.

Reply:
Thanks!